pysatMadrigal/instruments/methods/madrigal.py

```diff
@@ -58,114 +55,226 @@ def load(fnames, tag=None, inst_id=None, xarray_coords=[]):
+                                                        data.columns)]))
+
+                            # Remove the coordinates that aren't present
+                            temp = np.array(xarray_coords[xcoords])[data_mask]
+                            xarray_coords[xcoords] = list(temp)
+
+                    # Select the desired data values
+                    sel_data = data[list(xcoords) + xarray_coords[xcoords]]
+
+                    # Remove duplicates before indexing, to ensure data with
+                    # the same values at different locations are kept
+                    sel_data = sel_data.drop_duplicates()
+
+                    # Set the indices
+                    sel_data = sel_data.set_index(list(xcoords))
+
+                    # Recast as an xarray
+                    xdatasets.append(sel_data.to_xarray())
+
+                # Merge all of the datasets
+                for i in np.arange(1, len(xdatasets)):
+                    xdatasets[0] = xdatasets[0].merge(xdatasets[i])
+
+                # Test to see that all data was retrieved
+                test_variables = [xkey for xkey
+                                  in xdatasets[0].variables.keys()]
+                ltest = len(test_variables)
+                ldata = len(data.columns)
+
+                if ltest != ldata:
+                    if ltest < ldata:
+                        estr = 'missing: {:}'.format(
+                            ' '.join([dvar for dvar in data.columns
+                                      if dvar not in test_variables]))
+                    else:
+                        estr = 'have extra: {:}'.format(
+                            ' '.join([tvar for tvar in test_variables
+                                      if tvar not in data.columns]))
+                        raise ValueError(''.join(['coordinates not supplied ',
+                                                  'for all data columns: ',
+                                                  '{:d} != '.format(ltest),
+                                                  '{:d}; '.format(ldata),
+                                                  estr]))
+
+                data = xdatasets[0]
+            else:
+                # Set the index to time
+                data.index = time
+
+                # Raise a logging warning if there are duplicate times. This
+                # means the data should be stored as an xarray Dataset
+                if np.any(time.duplicated()):
+                    logger.warning(''.join(["duplicated time indices, consider",
+                                            " specifing additional coordinates",
+                                            " and storing the data as an ",
+                                            "xarray Dataset"]))
+        fdata.append(data)
+
+    # If multiple files were loaded, merge the data together
+    for i in np.arange(1, len(fdata)):
+        fdata[0] = fdata[0].merge(fdata[i])
+    data = fdata[0]
```

Comment:
Friendly reminder, may have been lost in translation. Based on pandas documentation doing a single concat is much faster than iteratively calling concat. While this isn't pandas it seems like a similar concern would apply. I went to do a small performance evaluation but multi-file support is in a different branch than this pull needs. Makes it hard to test as well :) 
```suggestion
        data = xr.merge(fdata)```
This works for a single file case.

The same suggestion technically goes for the hdf loads as well. Given just loading an hdf takes significant time the overhead may not be significant, percentage-wise, for that case.

Reply:
It isn't that simple, because the data may also be pandas.  But I incorporated your suggestion in a more appropriate way.